wp-parsely.php

```diff
@@ -46,3 +48,48 @@ function wpvip_load_wp_parsely_plugin() {
 	require 'wp-parsely-' . $major_version . '/wp-parsely.php';
 }
 add_action( 'after_setup_theme', 'wpvip_load_wp_parsely_plugin' );
+
+/**
+ * Pre-sets the "apikey" portion of the `parsely` site option hash map to the site_url IF it's empty.
+ * This is only hooked into `option_parsely` if we're loading the plugin from this repo.
+ *
+ * @param array $parsely_settings Settings from the database
+ * @return array Settings with the 'apikey' value overridden
+ */
+function wpvip_override_parsely_option_if_empty( $parsely_settings ) {
+	// Bail if an apikey is already set
+	if ( isset( $parsely_settings['apikey'] ) && strlen( $parsely_settings['apikey'] ) > 0 ) {
```

Comment:
```suggestion
	if ( isset( $parsely_settings['apikey'] ) && ! empty( $parsely_settings['apikey'] ) ) {
```

Reply:
Empty already allows for it not being set, so the first part of the condition could be dropped as well.

Alternatively, bearing in mind https://beberlei.de/2021/02/19/when_to_use_empty_in_php_i_say_never.html, then it's more prescriptive as to what we specifically want to check for, which is that it's set, and has a string length greater than 0.

Reply:
If we can drop the first condition, I think the code would look cleaner and simpler. However, I see your point on the article linked above, so either way works for me.

Reply:
I'm usually in the `empty` camp, but wrote it this way since I know Gary prefers the more prescriptive approach and I figured he'd have 👀 on this PR 😄.